src/core/Header/HeaderButton.tsx

```diff
@@ -57,6 +58,7 @@ export const HeaderButton = (props: HeaderButtonProps) => {
 
   useTheme();
 
+  const isSplitButton = menuItems && props.onClick;
   const buttonProps: ButtonProps & { styleType: 'borderless' } = {
```

Comment:
This type should be updated to include `menuItems`. And then you don't have to pass `menuItems` individually.